### ISSUE TITLE
Elastic search issue over Secure Connection

### ISSUE THREAD

**Issue by vineetagarwal1981** [magento/magento2#22297]
### Preconditions (*)
1. Magento Version 2.3
2. We are facing an issue while testing the connection of Elastic search over SSL without enable HTTP Auth. We are getting the response as Connection failed!.

### Steps to reproduce (*)

1. Magento Version 2.3
2. We have configuring AWS Elastic search with SSL.
3. We set the Elastic search detail in the backend Store > Configuration > Catalog > Catalog Search
4. Selected Elasticsearch 5.0+ and enter the hostname and port. HTTP Auth is not enabled.
5. Click on Test connection.
6. We are getting the response as Connection failed!.
```
	private function buildConfig($options = [])
    {
        $host = preg_replace('/http[s]?:\/\//i', '', $options['hostname']);
        $protocol = parse_url($options['hostname'], PHP_URL_SCHEME);
        if (!$protocol) {
            $protocol = 'http';
        }
        if (!empty($options['port'])) {
            $host .= ':' . $options['port'];
        }
        if (!empty($options['enableAuth']) && ($options['enableAuth'] == 1)) {
            $host = sprintf('%s://%s:%s@%s', $protocol, $options['username'], $options['password'], $host);
        }
		/*
			missing else condition if protocol is https
		*/
        $options['hosts'] = [$host];
        return $options;
    }
```

### Expected result (*)

1.  Connection should be successful.

### Actual result (*)
1. Connection failed!

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-backlog-nazar 
 Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-99154`, `MAGETWO-99155` were created

**Issue Available**: @engcom-backlog-nazar, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by atishgoswami** [comment]
I am working on this at #dmcdindia19

**Comment by hugodeaguiar** [comment]
Hello,

@atishgoswami Are you working on this problem yet?
We have this same problem.

Regards

**Comment by atishgoswami** [comment]
Hello @hugodeaguiar,

Yes I having a hard time to setup elasticsearch on local with a valid SSL.

Will try again this weekend, Will update soon.

Regards,
Atish

**Comment by magulin** [comment]
Hi @atishgoswami!
Any update regarding this?

**Comment by mariana-be** [comment]
@magento give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @mariana-be. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @mariana-be, here is your [Magento instance](https://i-22297-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-22297-2-3-develop.instances.magento-community.engineering/admin](https://i-22297-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by ihor-sviziev** [comment]
Here is fix for this issue: https://github.com/magento/magento2/pull/24636

**Comment by sdzhepa** [comment]
fixed in https://github.com/magento/magento2/pull/24636

**Comment by magento-engcom-team** [comment]
Hi @vineetagarwal1981. Thank you for your report. 
The issue has been fixed in magento/magento2#24636 by @ihor-sviziev in 2.3-develop branch
Related commit(s):
 - [b5bc8480ca7becfe4d1cb30b98866e9ff683bb5d](https://github.com/magento/magento2/commit/b5bc8480ca7becfe4d1cb30b98866e9ff683bb5d)
 - [dcdf3643a30612d879aa4d8ff4927782046e2f1e](https://github.com/magento/magento2/commit/dcdf3643a30612d879aa4d8ff4927782046e2f1e)
 - [2c46bd50269d99fbf3f1837b7ff8adc10cc8105d](https://github.com/magento/magento2/commit/2c46bd50269d99fbf3f1837b7ff8adc10cc8105d)
 - [e6adf506c1827e413e7bef255ebcb64b92d4372b](https://github.com/magento/magento2/commit/e6adf506c1827e413e7bef255ebcb64b92d4372b)
 - [f9f5d12b5228f4287424c469cc29d210a4279b0a](https://github.com/magento/magento2/commit/f9f5d12b5228f4287424c469cc29d210a4279b0a)

The fix will be available with the upcoming 2.3.4 release.
